File: src/test/java/org/dedda/bratwurst/lang/ReadVariableTest.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Created by dedda on 12/17/15.
 *
 * @author dedda
 */
public class ReadVariableTest {

    private Scope scope;

    @BeforeMethod
    public void setUp() throws Exception {
        Program program = new Program();
        scope = new Scope(program);
        scope.setVariable(new BWVariable("abc", new BWInteger(123)));
    }

    @Test
    public void testRun() throws Exception {
        ReadVariable readVariable = new ReadVariable(0, "abc");
        readVariable.run(scope);
        BWObject variable = readVariable.getValue();
        assertEquals(123, variable.getIntValue());
    }
}

File: src/test/java/org/dedda/bratwurst/lang/ReadVariableTest.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope
import org.testng.Assert
import org.testng.annotations.BeforeMethod
import org.testng.annotations.Test

/**
 * Created by dedda on 12/17/15.
 *
 * @author dedda
 */
class ReadVariableTest {
    private lateinit var scope: Scope

    @BeforeMethod
    @Throws(Exception::class)
    fun setUp() {
        val program = Program()
        scope = Scope(program)
        scope.setVariable(BWVariable("abc", BWInteger(123)))
    }

    @Test
    @Throws(Exception::class)
    fun testRun() {
        val readVariable = ReadVariable(0, "abc")
        readVariable.run(scope)
        val variable = readVariable.value
        Assert.assertEquals(123, variable.intValue)
    }
}